packages/ember-metal/lib/meta.js

```diff
@@ -104,6 +110,11 @@ Meta.prototype.isInitialized = function(obj) {
   return this.proto !== obj;
 };
 
+Meta.prototype.destroy = function() {
+  // clean up chain watchers
```

Comment:
It is being done when this is called in `ember-metal/watching`'s `destroy` function (which is the only place this is being called). I can move it here though if you'd prefer.